packages/flame/lib/src/components/position_component.dart

```diff
@@ -184,8 +185,16 @@ class PositionComponent extends Component
   /// This property can be reassigned at runtime, although this is not
   /// recommended. Instead, in order to make the [PositionComponent] larger
   /// or smaller, change its [scale].
+  @override
   NotifyingVector2 get size => _size;
-  set size(Vector2 size) => _size.setFrom(size);
+
+  @override
+  set size(Vector2 size) {
+    _size.setFrom(size);
+    if (hasChildren) {
```

Comment:
Does this check optimize anything? The foreach should also just do a check if it is empty right?

Reply:
`hasChildren` avoids instantiating a `ComponentSet` object in case it is null.

Reply:
could we do that check inside the `.children` getter instead?
that way we avoid duplicating this all over & everyone gets the performance boost (even end users)

Reply:
(can be a separate PR of course)

Reply:
The `.children` property returns a `ComponentSet` right now, which means it must create a component set in case it is currently missing. We could change it to return a `ComponentSet?` instead, but that would be a major breaking change I'm afraid -- so perhaps we could add that to our 2.0 todo list.

Reply:
Haha I think 3 people commented about this on this PR, maybe it would be better as nullable indeed, let's discuss that for V2 at least. :)

Reply:
I added a note about this in #1938